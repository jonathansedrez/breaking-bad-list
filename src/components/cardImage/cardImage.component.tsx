import React from 'react';

import { ICardImage } from './cardImage.types';
import './cardImage.styles.scss';

export const CardImage: React.FC<ICardImage> = (props: ICardImage) => {
  const { id, image, name, onClick } = props;

  return (
    <div
      className="card-image-wrapper"
      key={id}
      role="contentinfo"
      onClick={onClick}
    >
      <img src={image.src} alt={image.alt} className="card-image" />
      <div>
        <p className="card-image-text">{name}</p>
      </div>
    </div>
  );
};
