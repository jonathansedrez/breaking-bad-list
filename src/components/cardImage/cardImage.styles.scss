@import '../../layout/variables.scss';

.card-image-wrapper {
  width: 440px;
  height: 310px;
  position: relative;
  margin-bottom: 64px;
  cursor: pointer;
}

.card-image {
  width: 50%;
  height: 100%;
  border-radius: 2px;
}

.card-image-text {
  position: absolute;
  font-family: 'STIX Two Text', serif;
  color: $brown;
  top: 16px;
  left: 180px;
  width: 200px;
  font-size: 32px;
}
