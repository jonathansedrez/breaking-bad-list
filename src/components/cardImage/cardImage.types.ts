
export interface Image {
  src: string;
  alt: string;
}

export interface ICardImage {
  id: string;
  image: Image;
  name: string;
  onClick: () => void;
}