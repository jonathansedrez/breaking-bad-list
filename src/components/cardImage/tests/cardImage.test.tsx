import { render, fireEvent } from '@testing-library/react';

import { CardImage } from '../cardImage.component';

describe('List Characters screen', () => {
  it('should execute onpree when click an image', () => {
    const id = '123123';
    const name = 'John Doe';
    const image = {
      src: 'htttp://image.pnge',
      alt: 'alt image',
    };
    const onClick = jest.fn();

    const { getByRole, debug } = render(
      <CardImage id={id} onClick={onClick} image={image} name={name} />
    );
    const element = getByRole('contentinfo');

    expect(element).toBeInTheDocument();

    fireEvent.click(element);

    debug();
    expect(onClick).toBeCalled();
  });
});
