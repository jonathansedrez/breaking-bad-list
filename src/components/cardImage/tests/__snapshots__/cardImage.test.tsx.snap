// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Card Image should execute snapshot 1`] = `
<div
  class="card-image__wrapper"
  role="contentinfo"
>
  <img
    alt="alt image"
    class="card-image__image"
    src="htttp://image.pnge"
  />
  <div>
    <p
      class="card-image__text"
    >
      John Doe
    </p>
  </div>
</div>
`;
